### ISSUE TITLE
Performance Drop 0.9.0 vs 0.10.0

### ISSUE THREAD

**Issue by GoetzGoerisch** [hassio-addons/addon-tailscale#168]
# Problem/Motivation

Using HA with Tailscale as an exit node crippled the througput between `v0.9.0` and `v0.10.0`.

## Expected behavior

With version `0.9.0` one could achive about line speed, which is about 50MBit in my case with a raspberry-pi-4 using a smartphone and run all traffic through HA.

## Actual behavior

Speed is crippled to about 1Mbit.

## Steps to reproduce

- Advertise HA with Tailscale add-on as Exit Node
- Use exit node in mobil Tailscale app
- Run a speedtest with `0.9.0` and `0.10.0`

## Proposed changes

Investigate the impact of added taildrop on throughput.

**Comment by GoetzGoerisch** [comment]
Just ran a test with `v0.9.0` and `v0.10.0`.

Until about `-80 sec` in the graph `v0.9.0` of the add-on was running, from about `-65 sec` the `v0.10.0` was running. Downloading the same file.

![Screenshot 2023-03-31 17 01 56](https://user-images.githubusercontent.com/47734341/229157959-3f796cf6-d42b-4d27-849f-f08cdb61fa81.png)

Running speed tests shows a similar picture.

**Comment by frenck** [comment]
The only difference between those versions, is the addition of Taildrop support.

I've added an option in #185 to provide the option to disable it. Once shipped, that might be a good one to test disabled.

../Frenck

**Comment by GoetzGoerisch** [comment]
Not stale

**Comment by GoetzGoerisch** [comment]
@frenck any ETA when #185 is in a release?

**Comment by frenck** [comment]
Nope, there is a blocking issue that needs to be resolved first.

**Comment by GoetzGoerisch** [comment]
Not stale, wating for blocking issue to be resolved to test.

**Comment by lmagyar** [comment]
@GoetzGoerisch Meanwhile you can test #185 in my fork at: https://github.com/lmagyar/homeassistant-addon-tailscale (this is a drop in replacement until the official add-on is released, it contains a workaround for the blocking issue)

**Comment by GoetzGoerisch** [comment]
Could not test. Not stale

**Comment by GoetzGoerisch** [comment]
As per my recent tests with the forked extension, from @lmagyar, I see better results again.

Would prefer to leave it open until the upstream is merged.

**Comment by gabiginorio** [comment]
> 

Question here, did you just test with the forked extension default settings or did you set the option to "disable" the taildrop feature?

**Comment by gabiginorio** [comment]
Hi @GoetzGoerisch just following up to see if you had an answer to my above question.

Also, anyone know if there's a latest estimate for when the new release is coming out with these features?

**Comment by GoetzGoerisch** [comment]
@gabiginorio sorry for the delay. Left the default which is Taildrop OFF.

Tested with and without today. Taildrop still reduces throughout, but not as much as the 0.10.0 version.

**Comment by lmagyar** [comment]
> Left the default which is Taildrop OFF.

Just a note: by default taildrop is enabled! If there is no explicit `taildrop: false` line in the config, taildrop is enabled!

```
# Disable taildrop service when it is has been explicitly disabled
if bashio::config.false 'taildrop'; then
    rm /etc/s6-overlay/s6-rc.d/user/contents.d/taildrop
fi
```

**Comment by GoetzGoerisch** [comment]
Thanks @lmagyar! I tried it with explicitly swtiching Taildrop on and off.

Taildrop `on` shows less throughput but still better with the 0.10.0 default add-on.

**Comment by GoetzGoerisch** [comment]
Needs to be revised with the latest release.

**Comment by GoetzGoerisch** [comment]
Resolved with v0.13.1
